DigitalLearningSolutions.Web.Tests/Controllers/Register/ClaimAccountControllerTests.cs

```diff
@@ -399,204 +419,163 @@ public void
-            ).Returns(userEntity);
+                () => userDataService.GetUserIdAndCentreForCentreEmailRegistrationConfirmationHashPair(
+                    DefaultEmail,
+                    DefaultCode
+                )
+            ).Returns((null, null, null));
 
             // When
-            var result = controllerWithLoggedInUser.LinkDlsAccount();
+            var result = controllerWithLoggedInUser.LinkDlsAccountPost(DefaultEmail, DefaultCode);
 
             // Then
-            result.Should().BeRedirectToActionResult()
-                .WithActionName("AccountAlreadyExists");
+            A.CallTo(
+                () => userDataService.GetUserIdAndCentreForCentreEmailRegistrationConfirmationHashPair(
+                    DefaultEmail,
+                    DefaultCode
+                )
+            ).MustHaveHappenedOnceExactly();
 
-            controllerWithLoggedInUser.TempData.Peek<ClaimAccountViewModel>()!.Should().BeEquivalentTo(expectedModel);
+            result.Should().BeRedirectToActionResult().WithControllerName("LearningSolutions")
+                .WithActionName("AccessDenied");
         }
 
         [Test]
         public void
-            LinkDlsAccountPost_when_the_claim_email_address_matches_the_primary_email_of_another_user_redirects_to_WrongUser()
+            LinkDlsAccountPost_when_the_logged_in_user_already_has_a_delegate_account_at_the_centre_redirects_to_AccountAlreadyExists()
         {
             // Given
-            var expectedModel = GetViewModel(emailIsTaken: true);
-
-            controllerWithLoggedInUser.TempData.Set(expectedModel);
+            GivenValidViewModel(loggedInUserId: DefaultLoggedInUserId);
+            GivenUserEntity();
 
             // When
-            var result = controllerWithLoggedInUser.LinkDlsAccount();
+            var result = controllerWithLoggedInUser.LinkDlsAccountPost(DefaultEmail, DefaultCode);
 
             // Then
-            result.Should().BeRedirectToActionResult()
-                .WithActionName("WrongUser");
+            A.CallTo(() => userService.GetUserById(DefaultLoggedInUserId)).MustHaveHappenedOnceExactly();
 
-            controllerWithLoggedInUser.TempData.Peek<ClaimAccountViewModel>()!.Should().BeEquivalentTo(expectedModel);
+            result.Should().BeRedirectToActionResult().WithActionName("AccountAlreadyExists");
         }
 
         [Test]
         public void
-            LinkDlsAccountPost_with_valid_viewmodel_sets_expected_data_and_clears_temp_data_and_redirects_to_AccountsLinked()
+            LinkDlsAccountPost_when_the_claim_email_address_matches_the_primary_email_of_another_user_redirects_to_WrongUser()
         {
             // Given
-            var expectedModel = GetViewModel();
-
-            controllerWithLoggedInUser.TempData.Set(expectedModel);
+            GivenValidViewModel(loggedInUserId: DefaultLoggedInUserId, idOfUserMatchingEmailIfAny: DefaultUserId);

```

Comment:
Hardcoding the user ids - `GivenValidViewModel(loggedInUserId: 1, idOfUserMatchingEmailIfAny: 2);` - would make it even more obvious that one is different from the other and that's why you expect a "WrongUser" response.

Reply:
Done